pkgs/applications/editors/jetbrains/linux.nix

```diff
@@ -119,6 +123,18 @@ with stdenv; lib.makeOverridable mkDerivation (rec {
     runHook postInstall
   '';
 
+  postInstall = ''
+    if [[ "$pname" != "datagrip" && "$pname" != "dataspell" ]]; then
+      if [ -e "$out/$pname/bin/remote-dev-server.sh" ]; then
+        makeWrapper "$out/$pname/bin/remote-dev-server.sh" "$out/bin/$pname-remote-dev-server" \
+          --prefix PATH : "$out/libexec/$pname:${lib.makeBinPath [ jdk coreutils gnugrep which git ]}" \
+          --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath ([ stdenv.cc.cc.lib libsecret e2fsprogs libnotify ])}" \
+          --set-default JDK_HOME "${jdk}" \
+          --set-default JAVA_HOME "${jdk}"
+      fi
+    fi
+  '';
+
```

Comment:
Would it be possible to remove this if `$out/bin/clion` (the old place for the wrapper) was a symlink to `$out/clion/bin/clion`(the new place for the wrapper)?

Reply:
I'm not sure I understand the question. You mean removing the whole postInstall in that case?
I think it's needed not only for the location, but also for the env variables, but I might be wrong.

Reply:
No, I mean instead of wrapping `$out/clion/bin/clion` and putting the result in `$out/bin/clion`, move `$out/clion/bin/clion` to  `$out/clion/bin/.clion_wrapped`, put the wrapper at  `$out/clion/bin/clion` (so the remote ide launch script uses it) and make `$out/bin/clion` a symlink to `$out/clion/bin/clion`

Reply:
Well the thing I would be wrapping is not `clion` in this case, it would be `$out/clion/bin/remote-dev-server.sh`.
There are 2 executables that get installed `clion` and `clion-remote-dev-server`. 

`Clion` is the normal IDE in this case and is a binary, and `clion-remote-dev-server` is a script that sets up a bunch of variables and at the end calls `remote-dev-server.sh` which is what this is, wrapped.

If I just symlink clion to it, I lose the normal IDE, and I only get the remote extension functionality.

I might misunderstand your comment.

Reply:
The `remote-dev-server.sh` script calls the normal `$out/clion/bin/clion.sh` script, just with some extra args. I doubt any of the bash scripts need any of the programs added by the wrapper, so if we wrap `$out/clion/bin/clion.sh`, everything gets the extra programs in one place

Reply:
Are you saying that instead of having then two executables, the user will end up having just one, in this case Clion, that will be the wrapped script, and because both call the same initial script, if launched without the “run <path to project>” it could just run the IDE like `clion` runs it now, and if provided with those parameters it will run it like `clion-remote-dev-server` does it now?

Reply:
there will be two user-facing executables:
- `$out/bin/clion`: a symlink to `$out/clion/bin/clion.sh`, which is a wrapper script generated by us to add some stuff to the path, then call `$out/clion/bin/.clion_wrapped`
- `$out/bin/clion-remote-dev-server`: a symlink to `$out/clion/bin/remote-dev-server.sh`, which calls `$out/clion/plugins/remote-dev-server/bin/launcher.sh` (with some args), which calls  `$out/clion/bin/clion.sh` (with some args), which calls `$out/clion/bin/.clion_wrapped`

Reply:
I fully understand now. I didn't realise you were talking about the previous wrapper in the installPhase.

I have added a new commit removing the second wrapper, and I've tested it and it all still works great.

The reason why I created a wrapping script for the remote dev server as well is because the original script contains `$0` references that broke when I tried with a simple symlink, but now there is no more duplication about the makeWrapper variables!